src/inmanta/ast/type.py

```diff
@@ -1272,11 +1297,17 @@ def corresponds_to(self, type: Type) -> bool:
         return type.is_attribute_type()
 
     def issupertype(self, other: "Type") -> bool:
-        return other.is_attribute_type()
+        return other.is_attribute_type() and other != self
```

Comment:
Not sure about this, but it might be slightly better if is_attribute_type() is more costly performance wise ? (very marginal gain probably)
```suggestion
        return other != self and other.is_attribute_type()
```